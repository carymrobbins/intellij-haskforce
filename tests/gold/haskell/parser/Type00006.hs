module Type00006 where

newtype MaybeK a = MaybeK { runMaybeK :: forall r. (a -> Maybe r) -> Maybe r }
