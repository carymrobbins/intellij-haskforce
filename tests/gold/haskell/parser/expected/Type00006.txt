Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('Type00006')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    NEWTYPEDECL
      PsiElement(HaskellTokenType.newtype)('newtype')
      TYCON
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('MaybeK')
      TYVAR
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('a')
      PsiElement(HaskellTokenType.=)('=')
      NEWCONSTR
        CON
          CONID
            PsiElement(HaskellTokenType.conidRegexp)('MaybeK')
        ATYPE
          PsiElement(HaskellTokenType.{)('{')
          VARS
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('runMaybeK')
          PsiElement(HaskellTokenType.::)('::')
          TYPEE
            ATYPE
              PsiElement(HaskellTokenType.forall)('forall')
              TV_BNDR
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('r')
              PsiElement(HaskellTokenType..)('.')
              CTYPE
                TYPEE
                  ATYPE
                    PsiElement(HaskellTokenType.()('(')
                    CTYPE
                      TYPEE
                        ATYPE
                          TYVAR
                            VARID
                              PsiElement(HaskellTokenType.varidRegexp)('a')
                        PsiElement(HaskellTokenType.->)('->')
                        TYPEE
                          ATYPE
                            OQTYCON
                              QTYCON
                                TYCON
                                  CONID
                                    PsiElement(HaskellTokenType.conidRegexp)('Maybe')
                          ATYPE
                            TYVAR
                              VARID
                                PsiElement(HaskellTokenType.varidRegexp)('r')
                    PsiElement(HaskellTokenType.))(')')
                  PsiElement(HaskellTokenType.->)('->')
                  TYPEE
                    ATYPE
                      OQTYCON
                        QTYCON
                          TYCON
                            CONID
                              PsiElement(HaskellTokenType.conidRegexp)('Maybe')
                    ATYPE
                      TYVAR
                        VARID
                          PsiElement(HaskellTokenType.varidRegexp)('r')
          PsiElement(HaskellTokenType.})('}')